Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,344 @@
+             verboseLogging: verboseLogging
+         )
+         */
+
+        let builderImageName = baseImage
+
+        var builtProducts = [LambdaProduct: Path]()
+        for product in products {
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("building \"\(product.name)\" in docker")
+                print("-------------------------------------------------------------------------")
+            }
+            //
+            let buildCommand = "swift build --product \(product.name) -c \(buildConfiguration.rawValue) --static-swift-stdlib"
+            try self.execute(
+                executable: dockerToolPath,
+                arguments: ["run", "--rm", "-v", "\(packageDirectory.string):/workspace", "-w", "/workspace", builderImageName, "bash", "-cl", buildCommand],
+                verboseLogging: verboseLogging
+            )
+            #warning("this knows too much about the underlying implementation")
+            builtProducts[.init(product)] = packageDirectory.appending([".build", buildConfiguration.rawValue, product.name])
+        }
+        return builtProducts
+    }
+
+    private func build(
+        products: [Product],
+        buildConfiguration: PackageManager.BuildConfiguration,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        var results = [LambdaProduct: Path]()
+        for product in products {
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("building \"\(product.name)\"")
+                print("-------------------------------------------------------------------------")
+            }
+            var parameters = PackageManager.BuildParameters()
+            parameters.configuration = buildConfiguration
+            parameters.otherSwiftcFlags = ["-static-stdlib"]
+            parameters.logging = verboseLogging ? .verbose : .concise
+
+            let result = try packageManager.build(
+                .product(product.name),
+                parameters: parameters
+            )
+            guard result.builtArtifacts.count <= 1 else {
+                throw Errors.unknownExecutable("too many executable artifacts found for \(product.name)")
+            }
+            guard let artifact = result.builtArtifacts.first else {
+                throw Errors.unknownExecutable("no executable artifacts found for \(product.name)")
+            }
+            results[.init(product)] = artifact.path
+        }
+        return results
+    }
+
+    private func package(
+        products: [LambdaProduct: Path],
+        toolsProvider: (String) throws -> Path,
+        outputDirectory: Path,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        let zipToolPath = try toolsProvider("zip")
```

Comment:
If we do this, then we need to install zip in the AmazonLinux docker container first, since it should not be preinstalled.

Reply:
right. i wasn't to explore what using zlib would look like